lib/core/src/debug/monitor.rs

```diff
@@ -0,0 +1,603 @@
+    pub font_size             : Num,
+    pub font_vertical_offset  : Num,
+}
+
+pub type Config        = ConfigTemplate<String,u32>;
+pub type SamplerConfig = ConfigTemplate<JsValue,f64>;
+
+impl Default for Config {
+    fn default() -> Config {
+        Config {
+            background_color      : "#222222".into(),
+            label_color_ok        : "#8e939a".into(),
+            label_color_warn      : "#ffba18".into(),
+            label_color_err       : "#eb3941".into(),
+            plot_color_ok         : "#8e939a".into(),
+            plot_color_warn       : "#ffba18".into(),
+            plot_color_err        : "#eb3941".into(),
+            plot_background_color : "#333333".into(),
+            plot_step_size        : 1,
+            margin                : 4,
+            panel_height          : 15,
+            labels_width          : 120,
+            results_width         : 30,
+            plots_width           : 100,
+            font_size             : 9,
+            font_vertical_offset  : 4,
+        }
+    }
+}
+
+impl Config {
+    pub fn to_plot_config(&self) -> SamplerConfig {
+        let ratio      = window().device_pixel_ratio();
+        SamplerConfig {
+            background_color      : (&self.background_color)      . into(),
+            label_color_ok        : (&self.label_color_ok)        . into(),
+            label_color_warn      : (&self.label_color_warn)      . into(),
+            label_color_err       : (&self.label_color_err)       . into(),
+            plot_color_ok         : (&self.plot_color_ok)         . into(),
+            plot_color_warn       : (&self.plot_color_warn)       . into(),
+            plot_color_err        : (&self.plot_color_err)        . into(),
+            plot_background_color : (&self.plot_background_color) . into(),
+            plot_step_size        : self.plot_step_size       as f64 * ratio,
+            margin                : self.margin               as f64 * ratio,
+            panel_height          : self.panel_height         as f64 * ratio,
+            labels_width          : self.labels_width         as f64 * ratio,
+            results_width         : self.results_width        as f64 * ratio,
+            plots_width           : self.plots_width          as f64 * ratio,
+            font_size             : self.font_size            as f64 * ratio,
+            font_vertical_offset  : self.font_vertical_offset as f64 * ratio,
+        }
+    }
+}
+
+
+
+// ===============
+// === Monitor ===
+// ===============
+
+#[derive(Debug)]
+pub struct Monitor {
+    user_config   : Config,
+    config        : SamplerConfig,
+    width         : f64,
+    height        : f64,
+    dom           : Element,
+    panels        : Vec<Panel>,
+    canvas        : HtmlCanvasElement,
+    context       : CanvasRenderingContext2d,
+    is_first_draw : bool,
+}
+
+
+// === Public API ===
+
+// TODO: All the `unwraps` below should be handled nicer when this lib will be finished.
```

Comment:
I don't understand this todo. What lib, the whole basegl? Will basegl have the nicer handling of web_sys/js_sys errors?

Or if it's this "monitor" library, why we're trying to merge it if it's not finished?

Reply:
This is about "monitor" lib. Monitoring was not my task - it is task of @notdanilo , but I needed it to performance fine tune things, so I created initial implementation here.